<launch>
  <arg name="robot_name"/>
  <arg name="master_width"        default="1280"/>
  <arg name="master_height"        default="960"/>
  <arg name="align_depth"         default="true"/>
  <arg name="color_width"         default="$(arg master_width)"/>
  <arg name="color_height"        default="$(arg master_height)"/>
  <arg name="color_fps"         default="30"/>
  <arg name="depth_width"         default="$(arg master_width)"/>
  <arg name="depth_height"        default="$(arg master_height)"/>
  <arg name="depth_fps"         default="30"/>
  <arg name="infra_width"         default="$(arg master_width)"/>
  <arg name="infra_height"        default="$(arg master_height)"/>
  <arg name="infra_fps"         default="30"/>
  <arg name="enable_pointcloud"   default="true"/>
  <arg name="enable_infra1"   default="true"/>
  <arg name="enable_infra2"   default="true"/>
  <arg name="serial_no_camera"    			default="213622300470"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="camera"              			default="camera"/>		<!-- Note: Replace with camera name -->
  <arg name="tf_prefix_camera"         default="$(arg camera)"/>
  <arg name="publish_tf"         default="true"/>
  <arg name="initial_reset"             default="true"/>

<!-- D455 in 07-304 -->
  <group ns="$(arg camera)">
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
      <arg name="align_depth" value="$(arg align_depth)"/>
      <arg name="color_width" value="$(arg color_width)"/>
      <arg name="color_height" value="$(arg color_height)"/>
      <arg name="color_fps" value="$(arg color_fps)"/>
      <arg name="depth_width" value="$(arg depth_width)"/>
      <arg name="depth_height" value="$(arg depth_height)"/>
      <arg name="depth_fps" value="$(arg depth_fps)"/>
      <arg name="infra_width" value="$(arg infra_width)"/>
      <arg name="infra_height" value="$(arg infra_height)"/>
      <arg name="infra_fps" value="$(arg infra_fps)"/>
      <arg name="enable_pointcloud" value="$(arg enable_pointcloud)"/>
      <arg name="enable_infra1" value="$(arg enable_infra1)"/>
      <arg name="enable_infra2" value="$(arg enable_infra2)"/>
      <arg name="serial_no"             value="$(arg serial_no_camera)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera)"/>
      <arg name="publish_tf"         		value="$(arg publish_tf)"/>
      <arg name="enable_sync"         	value="true"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>

      <arg name="filters"               value="pointcloud"/> <!-- resolve problem of watchdog freeze etc. -->

    </include>
  </group>
</launch>